apps/management-app/src/components/Authenticated/CDO/WardsDetail/index.tsx

```diff
@@ -0,0 +1,170 @@
+  } = useForm<FormData>({
+    resolver: yupResolver(schema),
+    mode: 'onChange',
+    defaultValues: async () => {
+      const ward = await getWard(parseInt(id!), true).unwrap();
+      return ward
+        ? { name: ward.name, district_id: ward.district_id }
+        : { name: '', district_id: 0 };
+    },
+  });
+
+  const [updateWard] = useUpdateWardMutation();
+
+  const handleUpdateWard = useCallback(
+    handleSubmit(async (data: FormData) => {
+      try {
+        await updateWard({ id: parseInt(id!), data }).unwrap();
+        showSuccess('Ward updated');
+        reset(data);
+      } catch (error) {
+        showError(
+          isApiErrorResponse(error) ? error.data?.message : 'Unknown error',
+        );
+      }
+    }),
+    [updateWard],
+  );
+
+  const [deleteWards] = useDeleteWardsMutation();
+
+  const handleDeleteWard = useCallback(async () => {
+    try {
+      await deleteWards([parseInt(id!)]).unwrap();
+      showSuccess('Ward deleted');
+      navigate(-1);
+    } catch (error) {
+      showError(
+        isApiErrorResponse(error) ? error.data?.message : 'Unknown error',
+      );
+    }
+  }, [deleteWards, id, navigate]);
+
+  return (
+    <StaticActionBar
+      actionBarAlign="space-between"
+      actionBar={
+        <>
+          <Button
+            variant="outlined"
+            startIcon={<ArrowBackIcon />}
+            onClick={() => navigate(-1)}
+          >
+            Back
+          </Button>
+          <Box sx={{ display: 'inline-flex', columnGap: '16px' }}>
+            <Button
+              variant="outlined"
+              disabled={!isDirty}
+              onClick={() => reset()}
+            >
+              Reset
+            </Button>
+            <Button
+              variant="contained"
+              disabled={!isDirty || !isValid}
+              sx={{ color: (theme) => theme.palette.common.white }}
+              onClick={handleUpdateWard}
+            >
+              Save changes
+            </Button>
+            <Button
+              variant="contained"
+              color="error"
+              onClick={handleDeleteWard}
+            >
+              Delete
+            </Button>
+          </Box>
+        </>
+      }
+    >
+      <Box
+        component="form"
+        sx={{
+          width: '50%',
+          display: 'flex',
+          flexDirection: 'column',
+          rowGap: '16px',
+        }}
+      >
+        {isLoading || districtsLoading ? (
+          <>
+            <FormInputSkeleton label="Name" />
+            <FormInputSkeleton label="District" />
+          </>
+        ) : (
+          <>
+            <ControlledTextField control={control} name="name" label="Name" />
+            <ControlledSelect
+              control={control}
+              name="district_id"
+              label="District"
+              options={districts!.data!.map((e) => ({
+                value: e.id.toString(),
+                label: e.name,
+              }))}
```

Comment:
districts!.data, what does !. mean?

Reply:
Non-null assertion, changed. Now using districts value to determine render state instead.